apps/src/templates/instructions/TopInstructions.jsx

```diff
@@ -1,161 +1,28 @@
-  },
+  render() {
+    const { props } = this;
 
-  /**
-   * Given a prospective delta, determines how much we can actually change the
-   * height (accounting for min/max) and changes height by that much.
-   * @param {number} delta
-   * @returns {number} How much we actually changed
-   */
-  onHeightResize: function (delta) {
-    var minHeight = MIN_HEIGHT;
-    var currentHeight = this.props.height;
+    const ChildClass = props.csf ? TopInstructionsCSF : TopInstructionsCSP;
 
-    var newHeight = Math.max(minHeight, currentHeight + delta);
-    newHeight = Math.min(newHeight, this.props.maxHeight);
-
-    this.props.setInstructionsHeight(newHeight);
-    return newHeight - currentHeight;
-  },
-
-  componentDidMount: function () {
-    // Parent needs to readjust some sizing after images have loaded
-    $(ReactDOM.findDOMNode(this)).find('img').load(this.props.onLoadImage);
-  },
-
-  render: function () {
-    if (!this.props.markdown) {
-      return <div/>;
-    }
-    var id = this.props.id;
-
-    var mainStyle = [styles.main, {
-      height: this.props.height - RESIZER_HEIGHT
-    }, this.props.isEmbedView && styles.embedView];
-
-    return (
-      <div style={mainStyle} className="editor-column">
-        {!this.props.isEmbedView && <CollapserIcon
-            collapsed={this.props.collapsed}
-            onClick={this.props.toggleInstructionsCollapsed}/>
-        }
-        <div style={styles.header}>
-          {msg.puzzleTitle({
-            stage_total: this.props.stageTotal,
-            puzzle_number: this.props.puzzleNumber
-          })}
-        </div>
-        <div style={[this.props.collapsed && commonStyles.hidden]}>
-          <div style={styles.body}>
-            <Instructions
-              ref="instructions"
-              renderedMarkdown={processMarkdown(this.props.markdown)}
-              inTopPane
-              />
-          </div>
-          {!this.props.isEmbedView && <HeightResizer
-            position={this.props.height}
-            onResize={this.onHeightResize}/>
-          }
-        </div>
-      </div>
-    );
+    return <ChildClass ref="topInstructions" {...props}/>;
   }
 });
-module.exports = connect(function propsFromStore(state) {
-  return {
-    isEmbedView: state.pageConstants.isEmbedView,
-    puzzleNumber: state.pageConstants.puzzleNumber,
-    stageTotal: state.pageConstants.stageTotal,
-    maxHeight: state.instructions.maxHeight,
-    markdown: state.pageConstants.instructionsMarkdown,
-    collapsed: state.instructions.collapsed,
-  };
-}, function propsFromDispatch(dispatch) {
-  return {
-    toggleInstructionsCollapsed: function () {
-      dispatch(instructions.toggleInstructionsCollapsed());
-    },
-    setInstructionsHeight: function (height) {
-      dispatch(instructions.setInstructionsHeight(height));
-    }
-  };
-}, null, { withRef: true }
-)(Radium(TopInstructions));
+
+export default TopInstructions;
```

Comment:
Diff in this file is confusing, but basically I co-opted TopInstructions to be a wrapper class that gives you either a TopInstructionsCSF or TopInstructionsCSP. Once CSF instructions are more polished, it might be we can combine these back into one without too much difficulty